morf-postgresql/src/main/java/org/alfasoftware/morf/jdbc/postgresql/PostgreSQLDialect.java

```diff
@@ -934,4 +939,56 @@ protected String tableNameWithSchemaName(TableReference tableRef) {
       return tableRef.getDblink() + "." + tableRef.getName();
     }
   }
+
+  //
+  // ====== Auto-healing below ======
+  //
+
+  /**
+   * @see org.alfasoftware.morf.jdbc.SqlDialect#getSchemaConsistencyStatements(org.alfasoftware.morf.metadata.SchemaResource)
+   */
+  @Override
+  public List<String> getSchemaConsistencyStatements(SchemaResource schemaResource) {
+    return getPostgreSQLMetaDataProvider(schemaResource)
+            .map(this::getSchemaConsistencyStatements)
+            .orElseGet(() -> super.getSchemaConsistencyStatements(schemaResource));
+  }
+
+
+  private List<String> getSchemaConsistencyStatements(PostgreSQLMetaDataProvider metaDataProvider) {
+    return FluentIterable.from(metaDataProvider.tables())
+            .transformAndConcat(table -> healTable(metaDataProvider, table))
+            .toList(); // turn all the concatenated fluent iterables into a firm immutable list
+  }
+
+
+  private Iterable<String> healTable(PostgreSQLMetaDataProvider metaDataProvider, Table table) {
+    Iterable<String> statements = healIndexes(metaDataProvider, table);
+
+    if (statements.iterator().hasNext()) {
+      List<String> intro = ImmutableList.of(convertCommentToSQL("Auto-Healing table: " + table.getName()));
+      return Iterables.concat(intro, statements);
+    }
+    return ImmutableList.of();
+  }
+
+
+  private Iterable<String> healIndexes(PostgreSQLMetaDataProvider metaDataProvider, Table table) {
+    // Postgres 15 can deal with duplicate NULLs in unique indexes on it's own
+    if (Integer.parseInt(metaDataProvider.getDatabaseInformation().get(DatabaseMetaDataProvider.DATABASE_MAJOR_VERSION)) >= 15) {
+      // TODO
+      // See https://www.postgresql.org/docs/current/sql-createindex.html
+      // Once we support Postgres 15, we should introduce CREATE INDEX ... NULLS NOT DISTINCT
+      return ImmutableList.of();

```

Comment:
This is left here, as it will only make sense to use this once we decide to go ahead with fixing it.
At the moment, it serves as an example for other potential auto healers.